File: testing/test-data/password-checker.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4
# https://isaaccomputerscience.org/concepts/prog_softeng_test_plans 

def check_password(password):
    """Checks that a password length is between 8 and 14 characters. Returns True if valid"""
    is_valid = True
    if len(password) < 8 or len(password) > 14:
        is_valid = False
    return is_valid

File: testing/test-data/password-checker.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Microsoft Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{
    class Testing
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            
        }
        
        public bool CheckPassword (string password) {
            bool isValid = true;
            if (password.Length < 8 || password.Length > 14) {
                isValid = false;
            }
            return isValid;
        }
    }
}
